src/paperless_mail/mail.py

```diff
@@ -70,7 +71,11 @@ def get_criteria(self):
         return {"no_keyword": self.keyword}
 
     def post_consume(self, M: MailBox, message_uids, parameter):
-        M.flag(message_uids, [self.keyword], True)
+        if re.search(r"gmail\.com$|googlemail\.com$", M._host):
+            for uid in message_uids:
+                M.client.uid("STORE", uid, "X-GM-LABELS", self.keyword)
```

Comment:
I added a change (https://github.com/paperless-ngx/paperless-ngx/pull/1433/commits/1b56ee19ac33ee3e896c256bd151277c01eb4d8e) so that emails with labels will be ignored. I'm not sure if this is the nicest solution. And I have to add a test for it....

https://pypi.org/project/imap-tools/#search-criteria

Reply:
Glad to see imap_tools provides support for it.  The update's logic makes sense to me as well.

Reply:
New lines are covered by test. I had to extend some of the mocked Objects, so that they behave like imap_tools...